keras_nlp/models/bert/bert_classifier.py

```diff
@@ -190,6 +190,14 @@ def __init__(
         self.num_classes = num_classes
         self.dropout = dropout
 
+        # Default compilation
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(5e-5),
+            metrics=keras.metrics.SparseCategoricalAccuracy(),
+            jit_compile=True,
```

Comment:
This will blow up on TPU for all versions of TF pre 2.12, which isn't even out yet!.

I think we may need to carry some form of this change -> http://go/keras-team-keras/commit/1985e8acef8752eaeffa42c19b51341a6d91554b, and perhaps hide it away in a utility so it's not making our code gross everywhere.

cc @chenmoneygithub

Reply:
@mattdangerw all our current code re-compiles the models. There should be no test failure, right?

Reply:
We can have a condition check to set `jit_compile=True` if not using `TPUStrategy`.

Reply:
We shouldn't ship something that will break in it's default  usage on TPU though right? Our current code isn't the issue--the issue is that we don't want our new canonical quickstart to fail on TPU in a really obtuse way.

I'm pretty sure we want some sort of work around here unless we are planning to require tf 2.12 with the next release (which again is not even out yet).

Reply:
@chenmoneygithub can you share a pointer on how to check the strategy?